### PR TITLE
Phasing edits

### PR DESCRIPTION
**Pull request by guinneak198**

### CODE REVIEW

File: pyspecProcScripts/phasing.py [hunk 1]
```diff
@@ -1,12 +1,10 @@
 """This module includes routines for phasing NMR spectra."""
-from pyspecdata import *
-import time
+from pyspecdata import nddata, ndshape
 from matplotlib.patches import Ellipse
-from matplotlib.transforms import blended_transform_factory
```
Comment (guinneak198) [inline pyspecProcScripts/phasing.py:5]: not used

File: pyspecProcScripts/phasing.py [hunk 2]
```diff
@@ -1,12 +1,10 @@
 """This module includes routines for phasing NMR spectra."""
-from pyspecdata import *
-import time
```
Comment (guinneak198) [inline pyspecProcScripts/phasing.py:3]: not used

File: pyspecProcScripts/phasing.py [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
+from pyspecdata import nddata, ndshape
 from matplotlib.patches import Ellipse
-from matplotlib.transforms import blended_transform_factory
 from scipy.optimize import minimize
-from pylab import xlim, subplots, axvline, axhline, ylim, sca, rand, legend
```
Comment (guinneak198) [inline pyspecProcScripts/phasing.py:7]: nix functions not used

File: pyspecProcScripts/phasing.py [hunk 4]
```diff
@@ -107,7 +105,6 @@ def zeroth_order_ph(d, fl=None):
             evec_forplot[0, 0], evec_forplot[1, 0], "o", alpha=0.5, label="first evec"
         )
         fl.plot(evec_forplot[0, 1], evec_forplot[1, 1], "o", alpha=0.5)
-        norms = sqrt((evec_forplot**2).sum(axis=0))
```
Comment (guinneak198) [inline pyspecProcScripts/phasing.py:110]: unused variable

File: pyspecProcScripts/phasing.py [hunk 5]
```diff
@@ -166,6 +163,7 @@ def ph1_real_Abs(s, dw, ph1_sel=0, ph2_sel=1, fl=None):
     ph1_opt = np.asarray(s_cost.argmin("transients").item())
     logging.debug(strm("THIS IS PH1_OPT", ph1_opt))
     logging.debug(strm("optimal phase correction", repr(ph1_opt)))
+
```
Comment (guinneak198) [inline pyspecProcScripts/phasing.py:166]: black

File: pyspecProcScripts/phasing.py [hunk 6]
```diff
@@ -194,17 +192,58 @@ def costfun(ph1):
 
 def fid_from_echo(
     d,
-    signal_pathway,
+    signal_pathway=None,
     fl=None,
     add_rising=False,
-    fraction_nonnoise=0.1,
```
Comment (guinneak198) [inline pyspecProcScripts/phasing.py:200]: not used

File: pyspecProcScripts/phasing.py [hunk 7]
```diff
@@ -238,7 +277,24 @@ def filter_ranges(B, A):
...
+        #     just have "breaks" in the peak, so merge them.  Otherwise, fail.
+        if any(np.array(range_gaps) > max_range_width):
+            if fl is not None:
+                fl.next("debug filter ranges")
+                fl.plot(freq_envelope, human_units=False)
+                for thisrange in peakrange:
+                    fl.plot(freq_envelope[direct:thisrange], human_units=False)
+            raise ValueError("finding more than one peak!")
+        else:
+            peakrange = [(peakrange[0][0], peakrange[-1][1])]
```
Comment (guinneak198) [inline pyspecProcScripts/phasing.py:295]: we made this together to handle noisy data peaks

**Review by jmfranck — APPROVED** [review]

**Pull request merged**